protocols.py

```diff
@@ -1,67 +1,81 @@
 from binaryninja import (BinaryView, BackgroundTask, HighLevelILCall, RegisterValueType, HighLevelILAddressOf,
-                         HighLevelILVar, Constant, Function, HighLevelILVarSsa, HighLevelILVarInitSsa,
-                         TypeFieldReference, bundled_plugin_path, log_info, log_warn, log_alert)
-from typing import Optional, Tuple
+                         HighLevelILVar, Constant, Function, HighLevelILVarSsa, HighLevelILVarInitSsa, show_message_box,
+                         TypeFieldReference, bundled_plugin_path, log_info, log_warn, log_alert, user_directory)
+from binaryninja.enums import MessageBoxButtonSet
+from typing import Optional, Tuple, List
 import os
+import pathlib
+import json
 import sys
 import struct
 
-protocols = None
+protocols = {}
 
-def init_protocol_mapping():
-    # Parse EFI definitions only once
+
+def init_protocol_mapping(bv: BinaryView) -> bool:
+    """
+    Init protocol mappings, this function will parse bundled efi types and user provided types (if applicable)
+    If the user-provided mapping files contain errors, it will show a message box containing error position.
+    """
     global protocols
-    if protocols is not None:
-        return True
 
-    # Find the EFI type definition file within the Binary Ninja installation
+    current_path = pathlib.Path(__file__).parent
+    bundled_mapping_path = os.path.join(current_path, "data", "efi-protocol.json")
```

Comment:
Why are we bundling a `efi-protocol.json` in tree? What are these GUIDs for? Any GUIDs that we support natively should either be:

1. In our EFI platform types
2. Contributed to binarly's guiddb.json in their upstream repo (since we plan on pulling their file and bundling it)

We have too many sources for GUIDs. I'd like to limit it to our `efi.c` file, binarly's GUID DB, and for anything else the user wants, their own `efi-guids.json`